### PR TITLE
lib: use makeScopeWithSplicing' instead of makeScope

### PR DESCRIPTION
**Pull request by ipetkov**
## Motivation

Use a spliced scope that better allows for (more correctly) controlling what tools are used where when cross compiling.

Fixes https://github.com/ipetkov/crane/issues/628
Possibly related to https://github.com/ipetkov/crane/issues/551, though I still could not resolve that one yet

## Checklist

- [ ] added tests to verify new behavior
- [ ] added an example template or updated an existing one
- [ ] updated `docs/API.md` (or general documentation) with changes
- [ ] updated `CHANGELOG.md`

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -1,5 +1,13 @@
 { pkgs ? import <nixpkgs> { } }:
 
 import ./lib {
-  inherit (pkgs) lib newScope;
+  inherit (pkgs) lib makeScopeWithSplicing';
+  otherSplices = {
+    selfBuildBuild = pkgs.pkgsBuildBuild;
```
Comment (simonzkl) [inline default.nix:6]: I don't really understand how this solves the issue. `selfBuildBuild` should be `buildBuild` variants of crane's packages, not the ones from nixpkgs. Splicing `cargo` will just pick `cargo` from `pkgsBuildBuild`, same as before.

Comment (simonzkl) [reply default.nix:6]: Oh, actually I guess it does work if you pass a toolchain overlayed over nixpkgs.

Comment (ipetkov) [reply default.nix:6]: yes this was exactly the problem, using `craneLib.overrideToolchain` would use `.overrideScope` to define a new `cargo` definition, except that wouldn't make it into the spliced packages themselves, so the version from `nixpkgs` was being selected (which was still "wrong" even if spliced "correctly")

Comment (szlend) [reply default.nix:6]: Nope, this doesn't work. `callPackage` will splice `cargo` here as `pkgsBuildBuild.cargo`.

I got a few idea on how to fix this in #648.

**Pull request merged**